src/rpc/coinjoin.cpp

```diff
@@ -127,7 +130,10 @@ static RPCHelpMan coinjoin_start()
     }
 
     CHECK_NONFATAL(node.coinjoin_loader);
-    auto cj_clientman = node.coinjoin_loader->walletman().Get(wallet->GetName());
+    auto cj_clientman = [&]() {
+        LOCK(node.coinjoin_loader->walletman().cs_wallet_manager_map);
+        return node.coinjoin_loader->walletman().Get(wallet->GetName());
+    }();
```

Comment:
just use WITH_LOCK in these instances (even though I don't like this approach)

Reply:
I had done that but it's quite lengthy and it's functionally similar to what `WITH_LOCK` does (create a lambda expression and resolve it) 

```c++
auto cj_clientman = WITH_LOCK(node.coinjoin_loader->walletman().cs_wallet_manager_map, return node.coinjoin_loader->walletman().Get(wallet->GetName()));
```

Reply:
Current approach no longer does locking externally, resolved in latest push!